src/Entity/Ship.php

```diff
@@ -71,4 +71,23 @@ public function setGame(Game $game): self
         return $this;
     }
     
+    public function getPlayer(): ?Player
+    {
+        return $this->player;
+    }
+
+    public function setPlayer(Player $player): self
+    {
+        $this->player = $player;
+
+        return $this;
+    }
+
+    public function length() {
```

Comment:
Can you create a kind of "ShipManipulator" to handle `length()` and `isHorizontal()`? As I already said, Entity must be as simple as possible (only getters and setters)

Reply:
Even for this kind of methods ? Doing something like `$ship->isHorizontal()` looks much cleaner to me. Easier to read/understand. No ?

Reply:
It's a better not to mixing different programming styles. You use "functionnal" programming in your Manipulator and OOP programming here.

It makes the job so you can leave it for the moment. But can you add a description?
It seems that you've not setting up a linter yet? (you should do it at setup phase)